### ISSUE TITLE
[BUG] System.ComponentModel.Win32Exception: Access is denied.

### ISSUE THREAD

**Issue by Enzo5dj53** [bloxstraplabs/bloxstrap#4137]
### Acknowledgement of preliminary instructions

- [X] I have read the preliminary instructions, and I am certain that my problem has not already been addressed.
- [X] I have thoroughly looked through the available Wiki articles and could not find a solution to my problem.
- [X] I am using the latest version of Bloxstrap.
- [ ] I did not answer truthfully to all the above checkboxes.

### Bloxstrap Version

2.8.3

### What problem did you encounter?

launching roblox
"System.ComponentModel.Win32Exception: Access is denied."

### Bloxstrap Log

```text
2024-12-27T19:12:38Z [App::OnStartup] Starting Bloxstrap v2.8.3
2024-12-27T19:12:38Z [App::OnStartup] Compiled Friday, 27 December 2024 at 1:10:17 PM from commit a7e28200a9b14b94474dade8f0e7bb192e413067 (tag/v2.8.3)
2024-12-27T19:12:38Z [App::OnStartup] Loaded from %UserProfile%\AppData\Local\Bloxstrap\Bloxstrap.exe
2024-12-27T19:12:38Z [App::OnStartup] Temp path is %UserProfile%\AppData\Local\Temp\Bloxstrap
2024-12-27T19:12:38Z [App::OnStartup] WindowsStartMenu path is %UserProfile%\AppData\Roaming\Microsoft\Windows\Start Menu\Programs
2024-12-27T19:12:38Z [LaunchSettings] Identifier 'player' is active with data
2024-12-27T19:12:38Z [LaunchSettings] Invalid argument: roblox-player:1+launchmode:play+gameinfo:XkS3rv2D46gZoIXbGxDmoIEWD9aKDGHfuKAlyhN5LfhD9wA97nfCeWGLO96PnaFuBA4Dwd68IEjRCNaVTqezELI4foZCN657OGI9A37bDWa6JTKFMFZbFfDuGpwrnA8qImcLagLWEtm2mjLkGw94y_PAXfzpOq6f1Xj3T7taqhSHSYs6gqmsxs7e3sBBui42_eeUT2puaEU0aFLq0953uN_kLaM_wO-wg38ndEbUD5Q+launchtime:1735326757629+placelauncherurl:https%3A%2F%2Fwww.roblox.com%2FGame%2FPlaceLauncher.ashx%3Frequest%3DRequestPrivateGame%26browserTrackerId%3D1730581997598003%26placeId%3D1537690962%26accessCode%3D0e830ff8-785d-43d3-a8c4-8ed1c2b8c06f%26joinAttemptId%3D815514b0-cdaf-4a81-8afc-595d18e19112%26joinAttemptOrigin%3DprivateServerListJoin+browsertrackerid:1730581997598003+robloxLocale:en_us+gameLocale:en_us+channel:+LaunchExp:InApp
2024-12-27T19:12:38Z [LaunchSettings::ParsePlayer] Got Roblox launch arguments
2024-12-27T19:12:38Z [App::OnStartup] Entering main logic
2024-12-27T19:12:38Z [Logger::Initialize] Initializing at %UserProfile%\AppData\Local\Bloxstrap\Logs\Bloxstrap_20241227T191238Z.log
2024-12-27T19:12:38Z [Logger::Initialize] Finished initializing!
2024-12-27T19:12:38Z [JsonManager<Settings>::Load] Loading from %UserProfile%\AppData\Local\Bloxstrap\Settings.json...
2024-12-27T19:12:38Z [JsonManager<Settings>::Load] Loaded successfully!
2024-12-27T19:12:38Z [JsonManager<State>::Load] Loading from %UserProfile%\AppData\Local\Bloxstrap\State.json...
2024-12-27T19:12:38Z [JsonManager<State>::Load] Loaded successfully!
2024-12-27T19:12:38Z [FastFlagManager::Load] Loading from %UserProfile%\AppData\Local\Bloxstrap\Modifications\ClientSettings\ClientAppSettings.json...
2024-12-27T19:12:38Z [FastFlagManager::Load] Loaded successfully!
2024-12-27T19:12:38Z [LaunchHandler::ProcessLaunchArgs] Opening bootstrapper (Player)
2024-12-27T19:12:38Z [LaunchHandler::LaunchRoblox] Initializing bootstrapper
2024-12-27T19:12:38Z [LaunchHandler::LaunchRoblox] Initializing bootstrapper dialog
2024-12-27T19:12:38Z [Bootstrapper::Run] Running bootstrapper
2024-12-27T19:12:38Z [Bootstrapper::SetStatus] Connecting to Roblox...
2024-12-27T19:12:38Z [Deployment::TestConnection<https://setup.rbxcdn.com>] Connecting...
2024-12-27T19:12:38Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/versionStudio
2024-12-27T19:12:38Z [Deployment::InitializeConnectivity] Testing connectivity...
2024-12-27T19:12:38Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/versionStudio
2024-12-27T19:12:38Z [Deployment::InitializeConnectivity] Got https://setup.rbxcdn.com as the optimal base URL
2024-12-27T19:12:38Z [Bootstrapper::Run] Connectivity check finished
2024-12-27T19:12:38Z [Bootstrapper::CheckForUpdates] Checking for updates...
2024-12-27T19:12:38Z [HttpClientLoggingHandler::ProcessRequest] GET https://api.github.com/repos/bloxstraplabs/bloxstrap/releases/latest
2024-12-27T19:12:38Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://api.github.com/repos/bloxstraplabs/bloxstrap/releases/latest
2024-12-27T19:12:38Z [Bootstrapper::CheckForUpdates] No updates found
2024-12-27T19:12:38Z [Bootstrapper::GetLatestVersionInfo] Got channel as production
2024-12-27T19:12:38Z [Deployment::GetInfo] Getting deploy info for channel production
2024-12-27T19:12:38Z [HttpClientLoggingHandler::ProcessRequest] GET https://clientsettingscdn.roblox.com/v2/client-version/WindowsPlayer
2024-12-27T19:12:39Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://clientsettingscdn.roblox.com/v2/client-version/WindowsPlayer
2024-12-27T19:12:39Z [RegistryKeyEx::SetValueSafe] Writing '' to HKEY_CURRENT_USER\SOFTWARE\ROBLOX Corporation\Environments\RobloxPlayer\Channel\www.roblox.com
2024-12-27T19:12:39Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-b71c150c7c1f40de-rbxPkgManifest.txt
2024-12-27T19:12:39Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-b71c150c7c1f40de-rbxPkgManifest.txt
2024-12-27T19:12:39Z [Bootstrapper::SetStatus] Installing {product}...
2024-12-27T19:12:39Z [LaunchHandler::LaunchRoblox] Bootstrapper task has finished
2024-12-27T19:12:39Z [LaunchHandler::LaunchRoblox] An exception occurred when running the bootstrapper
2024-12-27T19:12:39Z [App::FinalizeExceptionHandling] (0x80004005) System.ComponentModel.Win32Exception (5): Access is denied.
   at System.Diagnostics.ProcessManager.OpenProcess(Int32 processId, Int32 access, Boolean throwIfExited)
   at System.Diagnostics.NtProcessManager.GetModules(Int32 processId, Boolean firstModuleOnly)
   at System.Diagnostics.NtProcessManager.GetFirstModule(Int32 processId)
   at Bloxstrap.Bootstrapper.KillRunningRobloxInDirectory(String path)
   at Bloxstrap.Bootstrapper.UpgradeRoblox()
   at Bloxstrap.Bootstrapper.Run()
   at Bloxstrap.Bootstrapper.Run()
2024-12-27T19:12:39Z [HttpClientLoggingHandler::ProcessRequest] POST https://bloxstraplabs.com/metrics/post-exception
```

**Comment by bluepilledgreat** [comment]
Fixed in v2.8.4

**Comment by Enzo5dj53** [comment]
thank you
